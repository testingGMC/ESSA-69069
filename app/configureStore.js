/**
 * Create the store with dynamic reducers
 */

import { createStore, applyMiddleware, compose } from 'redux'
import { routerMiddleware } from 'connected-react-router'
import createSagaMiddleware from 'redux-saga'
import { NOTIFICATION_TYPES } from 'shared/constants'
import { pushNotification } from 'shared/utils/notification'
import { errorsExtraction } from 'shared/utils/error-interceptors'
import reduxMiddlewares from './reduxMiddlewares'
import createReducer from './reducers'

export default function configureStore(initialState = {}, history) {
  let composeEnhancers = compose
  const reduxSagaMonitorOptions = {}

  // If Redux Dev Tools and Saga Dev Tools Extensions are installed, enable them
  /* istanbul ignore next */
  if (process.env.NODE_ENV !== 'production' && typeof window === 'object') {
    /* eslint-disable no-underscore-dangle */
    if (window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__)
      composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__({})

    // NOTE: Uncomment the code below to restore support for Redux Saga
    // Dev Tools once it supports redux-saga version 1.x.x
    // if (window.__SAGA_MONITOR_EXTENSION__)
    //   reduxSagaMonitorOptions = {
    //     sagaMonitor: window.__SAGA_MONITOR_EXTENSION__,
    //   };
    /* eslint-enable */
  }

  // const error400Handling = store => next => action => {
  //   if (action.e && action.type.slice(-7) === 'FAILURE') {
  //     if (action.e.response.status === 400) {
  //       const errors = errorsExtraction(action.e.response.data.model)
  //       errors.forEach(error =>
  //         pushNotification(NOTIFICATION_TYPES.error, error),
  //       )
  //       next({ ...action, errors })
  //       return
  //     }
  //   }
  //   next(action)
  // }

  const sagaMiddleware = createSagaMiddleware(reduxSagaMonitorOptions)

  // Create the store with two middlewares
  // 1. sagaMiddleware: Makes redux-sagas work
  // 2. routerMiddleware: Syncs the location/URL path to the state
  const middlewares = [
    sagaMiddleware,
    routerMiddleware(history),
    ...reduxMiddlewares,
  ]

  const enhancers = [applyMiddleware(...middlewares)]

  const store = createStore(
    createReducer(),
    initialState,
    composeEnhancers(...enhancers),
  )

  // Extensions
  store.runSaga = sagaMiddleware.run
  store.injectedReducers = {} // Reducer registry
  store.injectedSagas = {} // Saga registry

  // Make reducers hot reloadable, see http://mxs.is/googmo
  /* istanbul ignore next */
  if (module.hot) {
    module.hot.accept('./reducers', () => {
      store.replaceReducer(createReducer(store.injectedReducers))
    })
  }

  return store
}
