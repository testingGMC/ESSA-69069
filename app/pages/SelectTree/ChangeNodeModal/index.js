import { Form } from 'antd'
import ChangeNodeModal from './ChangeNodeModal'
export default Form.create({ name: 'change_node_modal' })(ChangeNodeModal)
