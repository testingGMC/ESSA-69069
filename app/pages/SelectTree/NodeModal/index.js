import { Form } from 'antd'
import NodeModal from './NodeModal'
export default Form.create({ name: 'node_modal' })(NodeModal)
