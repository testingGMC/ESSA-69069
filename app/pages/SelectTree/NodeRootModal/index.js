import { Form } from 'antd'
import NodeRootModal from './NodeRootModal'
export default Form.create({ name: 'node_root_modal' })(NodeRootModal)
