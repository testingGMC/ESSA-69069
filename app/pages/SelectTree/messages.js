/*
 * SelectTree Messages
 *
 * This contains all the text for the SelectTree component.
 */

import { defineMessages } from 'react-intl'

export const scope = 'app.components.SelectTree'

export default defineMessages({
  confirmMessage: {
    id: `${scope}.confirmMessage`,
    defaultMessage: 'Are you sure you want to delete this node',
  },
  cancel: {
    id: `${scope}.cancel`,
    defaultMessage: 'Cancel',
  },
  validate: {
    id: `${scope}.validate`,
    defaultMessage: 'Validate',
  },
  duplicatedTitleMessage: {
    id: `${scope}.title`,
    defaultMessage: 'You cannot use the same title twice',
  },
  duplicatedTitleDescription: {
    id: `${scope}.description`,
    defaultMessage: 'Every node must have a unique title',
  },
})
