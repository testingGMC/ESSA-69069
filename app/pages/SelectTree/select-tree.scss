.row-title {
  justify-content: center;
  align-items: center;
  margin-bottom: 27px;
  margin-left: 0px !important;
  margin-right: 0px !important;

  .ant-page-header {
      display: inline-flex;
      align-items: center;
      padding: 16px 4px;

      .ant-page-header-title-view-title {
          font-size: 28px;
      }
  }
}
.treehouse{
    display: flex;
    justify-content: center;
    .ant-empty{
      height: 50vh;
    }
    .ant-card-head {
      border-bottom: transparent !important;
    }
    .ant-card-body{
      padding-right: 0;
      padding-left: 0;
      padding-bottom: 0;
      .ant-tree{
        min-height: 50vh;
      }
      .divider{
        background-color: #e9edfb;
        display: flex;
        justify-content: center;
        height: 50px;
        .skillset{
          margin-top: 13px;
          margin-right: 50px;
          .skillimage{
            margin-right: 9px;
            margin-bottom: 2px;
            width: 18px;
          }
          
        }
        .button{
         background-color: #2a4bd8;
         border-color: #2a4bd8;
         margin-top: 7px;
        
        }
        .footer-button{
          margin-top: 5px;
          margin-left: 10px !important;
          background-color: transparent !important;
          border-color: transparent !important;
        }
        .no-show-button{
          display: none;
        }
      }
    }
    .card{
      width: calc(85%);
     }
  
  }
  .leaf{
    .nocase{
      display: none
    }
    .leaf-button{
      margin-left: 10px !important;
      background-color: transparent !important;
      border-color: transparent !important;
    }
  }
  .skill{
    color: #e74c3c;
  }
  .superskill{
    color: #3a9d23;
  }
  .track{
    color: #2980b9;
  }