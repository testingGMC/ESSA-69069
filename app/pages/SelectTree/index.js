/**
 *
 * SelectTree
 *
 */

import React, { memo } from 'react'
import { connect } from 'react-redux'

import { createStructuredSelector } from 'reselect'
import { compose } from 'redux'
import injectSaga from 'utils/injectSaga'
import { DAEMON } from 'utils/constants'
import { useInjectReducer } from 'utils/injectReducer'
import reducer from '../../shared/redux/selectTree/reducer'

import * as actions from '../../shared/redux/selectTree/actions'

import saga from '../../shared/redux/selectTree/saga'
import SelectTree from './SelectTree'
import makeSelectTrees from '../../shared/redux/selectTree/selectors'

const SelectTreeIndex = props => {
  useInjectReducer({ key: 'trees', reducer })
  return <SelectTree {...props} />
}

const mapStateToProps = () => ({
  trees: [],
})
const mapDispatchToProps = {
  ...actions,
}
const withSaga = injectSaga({ key: 'trees', saga, mode: DAEMON })

const withConnect = connect(
  mapStateToProps,
  mapDispatchToProps,
)

export default compose(
  withConnect,
  memo,
  withSaga,
)(SelectTreeIndex)
