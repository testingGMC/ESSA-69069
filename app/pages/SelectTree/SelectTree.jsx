import React, { useState, useEffect } from 'react'
import { FormattedMessage } from 'react-intl'
import {
  Tree,
  Card,
  Button,
  Row,
  Col,
  PageHeader,
  Empty,
  Popconfirm,
} from 'antd'
import { TYPES_TO_STRINGS, NOTIFICATION_TYPES, TYPES } from 'shared/constants'
import { pushNotification } from '../../shared/utils/notification'
import messages from './messages'
import NodeModal from './NodeModal'
import ChangeNodeModal from './ChangeNodeModal'
import NodeRootModal from './NodeRootModal'
import './select-tree.scss'

const { TreeNode } = Tree

const Leaf = ({ leaf, check, add, change, erase, type }) => (
  <div className={`leaf ${type}`}>
    {leaf}
    <span className={check ? 'showcase' : 'nocase'}>
      <Button
        size="small"
        icon="edit"
        shape="circle"
        className="leaf-button"
        onClick={change}
      />
      <Button
        size="small"
        icon="plus-circle"
        shape="circle"
        className="leaf-button"
        onClick={add}
      />
      <Popconfirm
        placement="topLeft"
        title={<FormattedMessage {...messages.confirmMessage} />}
        onConfirm={e => {
          e.stopPropagation(), erase()
        }}
        okText={<FormattedMessage {...messages.validate} />}
        cancelText={<FormattedMessage {...messages.cancel} />}
      >
        <Button
          size="small"
          icon="minus-circle"
          shape="circle"
          className="leaf-button"
          onClick={e => e.stopPropagation()}
        />
      </Popconfirm>
    </span>
  </div>
)
const SimpleLeaf = ({ leaf, type }) => (
  <div className={`simple-leaf ${type}`}>{leaf}</div>
)

const Footer = ({ add, publish, showcase }) => (
  <Row gutter={24}>
    <Col xs={24} sm={24} md={24} lg={24} xl={24}>
      <div className="divider">
        <div className="skillset">
          <img
            src={require('../../assets/icons/skill.png')}
            alt="skill"
            className="skillimage"
          />
          Skill
        </div>
        <div className="skillset">
          <img
            src={require('../../assets/icons/superskill.png')}
            alt="skill"
            className="skillimage"
          />
          Superskill
        </div>
        <div className="skillset">
          <img
            src={require('../../assets/icons/track.png')}
            alt="skill"
            className="skillimage"
          />
          Track
        </div>
        <Button type="primary" className="button" onClick={publish}>
          Publish
        </Button>
        <Button
          size="large"
          icon="plus-circle"
          shape="circle"
          className={`${showcase}`}
          onClick={add}
        />
      </div>
    </Col>
  </Row>
)

function SelectTree({ trees, fetchTrees, sendTree }) {
  const [expandedKeys] = useState([])
  const [gData, setGData] = useState([])
  const [checkedKeys, setKey] = useState([])
  const [checked, setCheck] = useState(false)
  const [theNodeType, setType] = useState(null)
  const [isVisible, setVisible] = useState(false)
  const [isChangeVisible, setChangeVisible] = useState(false)
  const [formRef, setFormRef] = useState(null)
  const [formReference, setFormReference] = useState(null)
  const [formRefr, setFormRefr] = useState(null)
  const [isAddVisible, setAddVisible] = useState(false)
  const [nodes, setNodes] = useState([])
  const [updatedNodeType, setUpdatedType] = useState(false)
  useEffect(() => {
    if (trees.data) {
      setGData(trees.data)
    }
  }, [trees])
  const onCheck = (checkedKey, e) => {
    const last = checkedKey.checked.slice(-1)[0]
    const lastEl = []
    lastEl.push(last)
    setKey(lastEl)
    setCheck(true)
    setType(
      e.checkedNodes.filter(el => el.key === last)[0].props.title.props.type,
    )
    checkNode(checkedKey)
  }
  const saveFormRef = formRefParameter => {
    setFormRef(formRefParameter)
  }
  const saveFormReference = formReferenceParameter => {
    setFormReference(formReferenceParameter)
  }
  const saveFormRefr = formRefrParameter => {
    setFormRefr(formRefrParameter)
  }
  const ModalClose = () => {
    const { form } = formRef.props
    form.resetFields()
    setVisible(false)
  }
  const ModalChangeClose = () => {
    const { form } = formReference.props
    form.resetFields()
    setChangeVisible(false)
  }
  const ModalAddClose = () => {
    const { form } = formRefr.props
    form.resetFields()
    setAddVisible(false)
  }
  const SuperSkillDisabled = type => {
    setUpdatedType(type)
  }
  const checkNode = key => {
    const loop = (data, key, callback) => {
      data.forEach(item => {
        if (item.children) {
          if (item.type === TYPES_TO_STRINGS[0]) {
            setUpdatedType(TYPES_TO_STRINGS[0])
          }
          if (item.children.map(el => el.type).includes(TYPES_TO_STRINGS[6])) {
            setUpdatedType(TYPES_TO_STRINGS[6])
          }
        }
        if (item.children) {
          return loop(item.children, key, callback)
        }
      })
    }
    const data = [...gData]
    loop(data, key, (item, index, arr) => {})
  }
  const changeNode = (key, title, description, type) => {
    let cNode
    const loop = (data, key, callback) => {
      const checkedNode = {
        title,
        name: title,
        key,
        type,
        description,
      }
      data.forEach(item => {
        if (item.children) {
          if (
            item.children.map(el => el.key).includes(checkedNode.key) &&
            item.type === 'skill'
          ) {
            SuperSkillDisabled(item.type)
          }
        }
        if (item.key === key) {
          cNode = item
          cNode.title = title
          cNode.description = description
          cNode.name = title
          cNode.type = type
          setNodes(nodes.map(node => (node.key === key ? cNode : node)))
        }
        if (item.children) {
          return loop(item.children, key, callback)
        }
      })
    }
    const data = [...gData]
    loop(data, key, (item, index, arr) => {})
    setGData(data)
    setChangeVisible(false)
  }

  const validateChange = key => {
    const { form } = formReference.props
    form.validateFields((err, values) => {
      if (err) {
        return
      }
      form.resetFields()
      const { nodeTitle, nodeDescription, nodeType } = values
      changeNode(key, nodeTitle, nodeDescription, nodeType)
    })
  }
  const isUnique = (rootNode, name) =>
    name !== rootNode.name &&
    (!rootNode.children ||
      rootNode.children.reduce(
        (res, next) => isUnique(next, name) && res,
        true,
      ))

  const validate = key => {
    const { form } = formRef.props
    form.validateFields((err, values) => {
      if (err) {
        return
      }
      if (!isUnique(gData[0], values.nodeTitle)) {
        pushNotification(
          NOTIFICATION_TYPES.error,
          <FormattedMessage {...messages.duplicatedTitleMessage} />,
          <FormattedMessage {...messages.duplicatedTitleDescription} />,
        )
      } else {
        form.resetFields()
        const { nodeTitle, nodeDescription, nodeType } = values
        nodeAdd(key, nodeTitle, nodeType, nodeDescription)
      }
    })
  }
  const validateAdd = () => {
    const { form } = formRefr.props
    form.validateFields((err, values) => {
      if (err) {
        return
      }
      form.resetFields()
      const { nodeTitle, nodeDescription, nodeType } = values
      addRoot(nodeTitle, nodeType, nodeDescription)
    })
  }
  const addRoot = (nodeTitle, nodeType, nodeDescription) => {
    const newKey = Math.floor(Math.random() * 1000200002).toString()
    setNodes([
      ...nodes,
      {
        title: nodeTitle,
        name: nodeTitle,
        key: newKey,
        type: nodeType,
        description: nodeDescription,
      },
    ])
    setGData([
      ...gData,
      {
        title: nodeTitle,
        name: nodeTitle,
        key: newKey,
        type: nodeType,
        description: nodeDescription,
      },
    ])
    setAddVisible(false)
  }

  const nodeAdd = (key, nodeTitle, nodeType, nodeDescription) => {
    let aNode

    const loop = (data, checkKey, callback) => {
      data.forEach(item => {
        if (item.title !== nodeTitle && item.key === checkKey) {
          aNode = item
          const newKey = `${checkKey}-${Math.floor(Math.random() * 1000200002)}`
          setNodes([
            ...nodes,
            {
              title: nodeTitle,
              name: nodeTitle,
              key: newKey,
              type: nodeType,
              description: nodeDescription,
            },
          ])

          if (!aNode.children && aNode.title !== nodeTitle) {
            aNode.children = []
            aNode.children.push({
              title: nodeTitle,
              name: nodeTitle,
              key: newKey,
              type: nodeType,
              description: nodeDescription,
            })
            checkNode(checkKey)
            onCheck('')
          } else if (aNode.title !== nodeTitle) {
            aNode.children.push({
              title: nodeTitle,
              name: nodeTitle,
              key: newKey,
              type: nodeType,
              description: nodeDescription,
            })
          }
        }
        if (item.children && item.title !== nodeTitle) {
          return loop(item.children, checkKey, callback)
        }
      })
    }
    setVisible(false)
    const data = [...gData]
    loop(data, key, (item, index, arr) => {})
    setGData(data)
  }
  const nodeRemove = key => {
    const loop = (data, key, callback) => {
      data.forEach(item => {
        if (item.key === key) {
          data.splice(data.indexOf(item), 1)
        }
        if (item.children) {
          return loop(item.children, key, callback)
        }
      })
    }
    const data = [...gData]
    loop(data, key, (item, index, arr) => {})
    setGData(data)
  }
  const onDragEnter = info => {}
  const onDrop = info => {
    const dropKey = info.node.props.eventKey
    const dragKey = info.dragNode.props.eventKey
    const dropPos = info.node.props.pos.split('-')
    const dropPosition = info.dropPosition - Number(dropPos[dropPos.length - 1])

    const loop = (data, key, callback) => {
      data.forEach((item, index, arr) => {
        if (item.key === key) {
          return callback(item, index, arr)
        }
        if (item.children) {
          return loop(item.children, key, callback)
        }
      })
    }
    const data = [...gData]

    let dragObj
    loop(data, dragKey, (item, index, arr) => {
      arr.splice(index, 1)
      dragObj = item
    })

    if (!info.dropToGap) {
      loop(data, dropKey, item => {
        item.children = item.children || []
        item.children.push(dragObj)
      })
    } else if (
      (info.node.props.children || []).length > 0 &&
      info.node.props.expanded &&
      dropPosition === 1
    ) {
      loop(data, dropKey, item => {
        item.children = item.children || []
        item.children.unshift(dragObj)
      })
    } else {
      let ar
      let i
      loop(data, dropKey, (item, index, arr) => {
        ar = arr
        i = index
      })
      if (dropPosition === -1) {
        ar.splice(i, 0, dragObj)
      } else {
        ar.splice(i + 1, 0, dragObj)
      }
    }

    setGData(data)
  }
  const loop = data =>
    data.map(item => {
      if (item.children && item.children.length) {
        return (
          <TreeNode
            title={
              checkedKeys.indexOf(item.key) !== -1 ? (
                <Leaf
                  leaf={item.title}
                  erase={() => nodeRemove(checkedKeys[0])}
                  add={() => setVisible(true)}
                  change={() => setChangeVisible(true)}
                  check={checked}
                />
              ) : (
                <SimpleLeaf type={item.type} leaf={item.title} />
              )
            }
            key={item.key}
          >
            {loop(item.children)}
          </TreeNode>
        )
      }
      return (
        <TreeNode
          key={item.key}
          title={
            checkedKeys.indexOf(item.key) !== -1 ? (
              <Leaf
                type={item.type}
                leaf={item.title}
                erase={() => nodeRemove(checkedKeys[0])}
                add={() => setVisible(true)}
                change={() => setChangeVisible(true)}
                check={checked}
              />
            ) : (
              <SimpleLeaf type={item.type} leaf={item.title} />
            )
          }
        />
      )
    })

  const findNodeByKey = key => nodes.find(node => node.key === key)
  return (
    <Row gutter={24} type="flex" className="row-title">
      <Col span={20}>
        <PageHeader
          onBack={() => {
            window.history.back()
          }}
          title="Tree"
        />
      </Col>
      <Col xs={24} sm={24} md={12} lg={24} xl={24}>
        <div className="treehouse">
          <Card title="Folders" className="card">
            {gData.length === 0 ? (
              <Empty />
            ) : (
              <Tree
                className="draggable-tree"
                defaultExpandedKeys={expandedKeys}
                draggable
                blockNode
                onDragEnter={onDragEnter}
                onDrop={onDrop}
                checkable
                checkStrictly
                onCheck={onCheck}
              >
                {loop(gData)}
              </Tree>
            )}

            {gData.length === 0 ? (
              <Footer
                showcase="footer-button"
                add={() => setAddVisible(true)}
                publish={() => sendTree(gData)}
              />
            ) : (
              <Footer
                showcase="no-show-button"
                add={() => setAddVisible(true)}
                publish={() => sendTree(gData)}
              />
            )}
          </Card>
          <NodeModal
            wrappedComponentRef={saveFormRef}
            isVisible={isVisible}
            handleCancel={ModalClose}
            handleOk={() => validate(checkedKeys[0])}
            disabled={theNodeType === TYPES_TO_STRINGS[0]}
          />
          <ChangeNodeModal
            wrappedComponentRef={saveFormReference}
            isVisible={isChangeVisible}
            handleCancel={ModalChangeClose}
            handleOk={() => validateChange(checkedKeys[0])}
            node={findNodeByKey(checkedKeys[0])}
            disabled={updatedNodeType === TYPES_TO_STRINGS[0]}
            superDisabled={updatedNodeType === TYPES_TO_STRINGS[6]}
          />
          <NodeRootModal
            wrappedComponentRef={saveFormRefr}
            isVisible={isAddVisible}
            handleCancel={ModalAddClose}
            handleOk={() => validateAdd(checkedKeys[0])}
          />
        </div>
      </Col>
    </Row>
  )
}

export default SelectTree
