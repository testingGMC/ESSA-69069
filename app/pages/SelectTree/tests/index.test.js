import React from 'react'
import { render } from 'react-testing-library'
import { IntlProvider } from 'react-intl'
import { Provider } from 'react-redux'
import { browserHistory } from 'react-router-dom'

import SelectTreeIndex from '../index'
import SelectTree from '../SelectTree'
import { DEFAULT_LOCALE } from '../../../i18n'
import configureStore from '../../../configureStore'

const initialProps = {
  fetchTrees: jest.fn(),
  sendTree: jest.fn(),
  trees: {}
}

describe('<SelectTreeIndex />', () => {
  let store
  beforeAll(() => {
    store = configureStore({}, browserHistory)
  })
  it('Expect to not log in console', () => {
    const spy = jest.spyOn(global.console, 'log')
    render(
      <Provider store={store}>
        <IntlProvider locale={DEFAULT_LOCALE}>
          <SelectTreeIndex {...initialProps} />
        </IntlProvider>
      </Provider>
    )
    expect(spy).not.toHaveBeenCalled()
  })


  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error')
    render(
      <Provider store={store}>
        <IntlProvider locale={DEFAULT_LOCALE}>
          <SelectTreeIndex {...initialProps} />
        </IntlProvider>
      </Provider>,
    )
    expect(spy).not.toHaveBeenCalled()
  })
})

describe('<SelectTree/>', () => {
  let store
  beforeAll(() => {
    store = configureStore({}, browserHistory)
  })

  it('should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <Provider store={store}>
        <IntlProvider locale={DEFAULT_LOCALE}>
          <SelectTree {...initialProps} />
        </IntlProvider>
      </Provider>,
    )
    expect(firstChild).toMatchSnapshot()
  })

  it('should fetch the trees on mount if the tree exists', () => {
    render(
      <Provider store={store}>
        <IntlProvider locale={DEFAULT_LOCALE}>
          <SelectTree {...initialProps} trees />
        </IntlProvider>
      </Provider>,
    )
    expect(initialProps.fetchTrees).not.toHaveBeenCalled()
  })
})