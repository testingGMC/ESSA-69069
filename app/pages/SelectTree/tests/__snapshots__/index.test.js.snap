// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SelectTree/> should render and match the snapshot 1`] = `
<div
  class="ant-row-flex row-title"
  style="margin-left: -12px; margin-right: -12px;"
>
  <div
    class="ant-col ant-col-20"
    style="padding-left: 12px; padding-right: 12px;"
  >
    <div
      class="ant-page-header"
    >
      <div
        class="ant-page-header-back"
      >
        <div
          aria-label="back"
          class="ant-page-header-back-button"
          role="button"
          style="border: 0px; background: transparent; padding: 0px; line-height: inherit; display: inline-block;"
          tabindex="0"
        >
          <i
            aria-label="icon: arrow-left"
            class="anticon anticon-arrow-left"
          >
            <svg
              aria-hidden="true"
              class=""
              data-icon="arrow-left"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M872 474H286.9l350.2-304c5.6-4.9 2.2-14-5.2-14h-88.5c-3.9 0-7.6 1.4-10.5 3.9L155 487.8a31.96 31.96 0 0 0 0 48.3L535.1 866c1.5 1.3 3.3 2 5.2 2h91.5c7.4 0 10.8-9.2 5.2-14L286.9 550H872c4.4 0 8-3.6 8-8v-60c0-4.4-3.6-8-8-8z"
              />
            </svg>
          </i>
        </div>
        <div
          class="ant-divider ant-divider-vertical"
        />
      </div>
      <div
        class="ant-page-header-title-view"
      >
        <span
          class="ant-page-header-title-view-title"
        >
          Tree
        </span>
      </div>
    </div>
  </div>
  <div
    class="ant-col ant-col-xs-24 ant-col-sm-24 ant-col-md-12 ant-col-lg-24 ant-col-xl-24"
    style="padding-left: 12px; padding-right: 12px;"
  >
    <div
      class="treehouse"
    >
      <div
        class="ant-card card ant-card-bordered"
      >
        <div
          class="ant-card-head"
        >
          <div
            class="ant-card-head-wrapper"
          >
            <div
              class="ant-card-head-title"
            >
              Folders
            </div>
          </div>
        </div>
        <div
          class="ant-card-body"
        >
          <div
            class="ant-empty"
          >
            <div
              class="ant-empty-image"
            >
              <img
                alt="No Data"
                src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTg0IiBoZWlnaHQ9IjE1MiIgdmlld0JveD0iMCAwIDE4NCAxNTIiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyI+CiAgPGcgZmlsbD0ibm9uZSIgZmlsbC1ydWxlPSJldmVub2RkIj4KICAgIDxnIHRyYW5zZm9ybT0idHJhbnNsYXRlKDI0IDMxLjY3KSI+CiAgICAgIDxlbGxpcHNlIGZpbGwtb3BhY2l0eT0iLjgiIGZpbGw9IiNGNUY1RjciIGN4PSI2Ny43OTciIGN5PSIxMDYuODkiIHJ4PSI2Ny43OTciIHJ5PSIxMi42NjgiLz4KICAgICAgPHBhdGggZD0iTTEyMi4wMzQgNjkuNjc0TDk4LjEwOSA0MC4yMjljLTEuMTQ4LTEuMzg2LTIuODI2LTIuMjI1LTQuNTkzLTIuMjI1aC01MS40NGMtMS43NjYgMC0zLjQ0NC44MzktNC41OTIgMi4yMjVMMTMuNTYgNjkuNjc0djE1LjM4M2gxMDguNDc1VjY5LjY3NHoiIGZpbGw9IiNBRUI4QzIiLz4KICAgICAgPHBhdGggZD0iTTEwMS41MzcgODYuMjE0TDgwLjYzIDYxLjEwMmMtMS4wMDEtMS4yMDctMi41MDctMS44NjctNC4wNDgtMS44NjdIMzEuNzI0Yy0xLjU0IDAtMy4wNDcuNjYtNC4wNDggMS44NjdMNi43NjkgODYuMjE0djEzLjc5Mmg5NC43NjhWODYuMjE0eiIgZmlsbD0idXJsKCNsaW5lYXJHcmFkaWVudC0xKSIgdHJhbnNmb3JtPSJ0cmFuc2xhdGUoMTMuNTYpIi8+CiAgICAgIDxwYXRoIGQ9Ik0zMy44MyAwaDY3LjkzM2E0IDQgMCAwIDEgNCA0djkzLjM0NGE0IDQgMCAwIDEtNCA0SDMzLjgzYTQgNCAwIDAgMS00LTRWNGE0IDQgMCAwIDEgNC00eiIgZmlsbD0iI0Y1RjVGNyIvPgogICAgICA8cGF0aCBkPSJNNDIuNjc4IDkuOTUzaDUwLjIzN2EyIDIgMCAwIDEgMiAyVjM2LjkxYTIgMiAwIDAgMS0yIDJINDIuNjc4YTIgMiAwIDAgMS0yLTJWMTEuOTUzYTIgMiAwIDAgMSAyLTJ6TTQyLjk0IDQ5Ljc2N2g0OS43MTNhMi4yNjIgMi4yNjIgMCAxIDEgMCA0LjUyNEg0Mi45NGEyLjI2MiAyLjI2MiAwIDAgMSAwLTQuNTI0ek00Mi45NCA2MS41M2g0OS43MTNhMi4yNjIgMi4yNjIgMCAxIDEgMCA0LjUyNUg0Mi45NGEyLjI2MiAyLjI2MiAwIDAgMSAwLTQuNTI1ek0xMjEuODEzIDEwNS4wMzJjLS43NzUgMy4wNzEtMy40OTcgNS4zNi02LjczNSA1LjM2SDIwLjUxNWMtMy4yMzggMC01Ljk2LTIuMjktNi43MzQtNS4zNmE3LjMwOSA3LjMwOSAwIDAgMS0uMjIyLTEuNzlWNjkuNjc1aDI2LjMxOGMyLjkwNyAwIDUuMjUgMi40NDggNS4yNSA1LjQydi4wNGMwIDIuOTcxIDIuMzcgNS4zNyA1LjI3NyA1LjM3aDM0Ljc4NWMyLjkwNyAwIDUuMjc3LTIuNDIxIDUuMjc3LTUuMzkzVjc1LjFjMC0yLjk3MiAyLjM0My01LjQyNiA1LjI1LTUuNDI2aDI2LjMxOHYzMy41NjljMCAuNjE3LS4wNzcgMS4yMTYtLjIyMSAxLjc4OXoiIGZpbGw9IiNEQ0UwRTYiLz4KICAgIDwvZz4KICAgIDxwYXRoIGQ9Ik0xNDkuMTIxIDMzLjI5MmwtNi44MyAyLjY1YTEgMSAwIDAgMS0xLjMxNy0xLjIzbDEuOTM3LTYuMjA3Yy0yLjU4OS0yLjk0NC00LjEwOS02LjUzNC00LjEwOS0xMC40MDhDMTM4LjgwMiA4LjEwMiAxNDguOTIgMCAxNjEuNDAyIDAgMTczLjg4MSAwIDE4NCA4LjEwMiAxODQgMTguMDk3YzAgOS45OTUtMTAuMTE4IDE4LjA5Ny0yMi41OTkgMTguMDk3LTQuNTI4IDAtOC43NDQtMS4wNjYtMTIuMjgtMi45MDJ6IiBmaWxsPSIjRENFMEU2Ii8+CiAgICA8ZyB0cmFuc2Zvcm09InRyYW5zbGF0ZSgxNDkuNjUgMTUuMzgzKSIgZmlsbD0iI0ZGRiI+CiAgICAgIDxlbGxpcHNlIGN4PSIyMC42NTQiIGN5PSIzLjE2NyIgcng9IjIuODQ5IiByeT0iMi44MTUiLz4KICAgICAgPHBhdGggZD0iTTUuNjk4IDUuNjNIMEwyLjg5OC43MDR6TTkuMjU5LjcwNGg0Ljk4NVY1LjYzSDkuMjU5eiIvPgogICAgPC9nPgogIDwvZz4KPC9zdmc+Cg=="
              />
            </div>
            <p
              class="ant-empty-description"
            >
              No Data
            </p>
          </div>
          <div
            class="ant-row"
            style="margin-left: -12px; margin-right: -12px;"
          >
            <div
              class="ant-col ant-col-xs-24 ant-col-sm-24 ant-col-md-24 ant-col-lg-24 ant-col-xl-24"
              style="padding-left: 12px; padding-right: 12px;"
            >
              <div
                class="divider"
              >
                <div
                  class="skillset"
                >
                  <img
                    alt="skill"
                    class="skillimage"
                    src="IMAGE_MOCK"
                  />
                  Skill
                </div>
                <div
                  class="skillset"
                >
                  <img
                    alt="skill"
                    class="skillimage"
                    src="IMAGE_MOCK"
                  />
                  Superskill
                </div>
                <div
                  class="skillset"
                >
                  <img
                    alt="skill"
                    class="skillimage"
                    src="IMAGE_MOCK"
                  />
                  Track
                </div>
                <button
                  class="ant-btn button ant-btn-primary"
                  type="button"
                >
                  <span>
                    Publish
                  </span>
                </button>
                <button
                  class="ant-btn footer-button ant-btn-circle ant-btn-lg ant-btn-icon-only"
                  type="button"
                >
                  <i
                    aria-label="icon: plus-circle"
                    class="anticon anticon-plus-circle"
                  >
                    <svg
                      aria-hidden="true"
                      class=""
                      data-icon="plus-circle"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M696 480H544V328c0-4.4-3.6-8-8-8h-48c-4.4 0-8 3.6-8 8v152H328c-4.4 0-8 3.6-8 8v48c0 4.4 3.6 8 8 8h152v152c0 4.4 3.6 8 8 8h48c4.4 0 8-3.6 8-8V544h152c4.4 0 8-3.6 8-8v-48c0-4.4-3.6-8-8-8z"
                      />
                      <path
                        d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
                      />
                    </svg>
                  </i>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
